views.py

```diff
@@ -1,47 +1,86 @@
 from flask import Blueprint, render_template, request, redirect, url_for
+import json
+import os
+
 
 main = Blueprint('main', __name__)
 
-items = []
+
 @main.route('/')
 def index():
-	return render_template('homepage.html', items=items)
+	#with open('list.json') as file:
+	#	items = json.load(file)
+	with open('lists.json') as file:
+		try:
+			lists = json.load(file)
+		except:
+			lists = ''
+	return render_template('homepage.html', lists=lists)
 
-@main.route('/list')
+@main.route('/list', methods=['POST'])
 def list():
-	return render_template('list.html', items=items)
-
-@main.route('/remove', methods=['POST'])
-def list_remove():
-	list_remove = request.form.get('removes')
-	if list_remove:
-		items.remove(f'{list_remove}')
-	return render_template('homepage.html', items=items)
+	name = request.form.get('chooselist')
+	with open('/Users/austin/src/web_app/lists/' + name + '.json') as file:
+		items = json.load(file)
+	return render_template('list.html', items=items, name=name)
 
 @main.route('/list_post', methods=['POST'])
 def list_post():
+	name = request.form.get('names')
+	with open('/Users/austin/src/web_app/lists/' + name + '.json') as file:
+		items = json.load(file)
```

Comment:
This stuff is re-used a lot.

You could have a generic function to load your files by name.

```
def getItems(name):
    with open('/Users/austin/src/web_app/lists/' + name + '.json') as file:
		return json.load(file)
```

Then you don't have to re-do this every time. It'd just be
```
items = getItems(name)
```

And you can re-use this function everywhere you're loading items below. And it's more readable!

Also - will this die if the file doesn't exist?

Reply:
good idea on the function so i don't have to retype it and with t he relative pathing. yes it would die if it didn't exist, but for it not to exist, you'd have to do some fuckity shit. because the variable name is set by 'names' which is actually a hidden html drop down which stores the variable from when you select it from line 21, def list():.